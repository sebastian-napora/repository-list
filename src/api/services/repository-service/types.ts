import gql from 'graphql-tag';

export type TGetRepostioryService = {
  get(): ReturnType<typeof gql>;
  getBySearchingPhrase(): ReturnType<typeof gql>;
};
