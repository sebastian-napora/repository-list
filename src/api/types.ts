import { getRepostioryService } from './services/repository-service';

import { GET_REPOSITORY_SERVICE_KEY } from '../utils/constants';

export type TGetServices = {
  [GET_REPOSITORY_SERVICE_KEY]: typeof getRepostioryService;
};

export type TGetServicesNames = keyof TGetServices;

//Object services types
export type TGetServicesKeys = {
  [Key in TGetServicesNames]: TGetServices[Key];
};
