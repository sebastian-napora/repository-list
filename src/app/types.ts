import { TSingleTheme } from '../theme/types';

export type TUseApp = () => {
  selectedTheme: TSingleTheme;
  themeLoaded: boolean;
};

export type THandleSelectedTheme = () => void;
