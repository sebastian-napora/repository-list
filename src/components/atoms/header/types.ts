import { FC } from 'react';

export type THeaderProps = FC<{
  title: string;
}>;
