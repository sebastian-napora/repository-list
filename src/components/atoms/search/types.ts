import { ChangeEvent, FC } from 'react';

export type TSearchBarProps = FC<{
  label: string;
  onChange?: ((event: ChangeEvent<HTMLInputElement>) => void) | undefined;
}>;
