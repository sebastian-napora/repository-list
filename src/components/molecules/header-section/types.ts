import { FC } from 'react';

export type THeaderSectionProps = FC<{
  title: string;
  children: JSX.Element;
}>;
