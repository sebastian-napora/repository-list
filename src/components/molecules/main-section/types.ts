import { FC } from 'react';

export type TMainSectionProps = FC<{
  children: JSX.Element;
}>;
