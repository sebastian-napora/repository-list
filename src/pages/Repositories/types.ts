import { ApolloError } from '@apollo/client';
import { IRow, TRepostioriesQueryData } from '../../core/typings/repositories';

export type TUseRepositories = () => {
  loading: boolean;
  error: ApolloError | undefined;
  rows: IRow[];
};

export type TGetNewRowsData = (data: TRepostioriesQueryData) => void;

export type TCreateNewRowsDataArray = (data: TRepostioriesQueryData) => {
  name: string;
  stars: number;
  forks: number;
  id: string;
  url: string;
}[];
