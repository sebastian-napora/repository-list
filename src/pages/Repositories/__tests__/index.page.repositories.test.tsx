import React from 'react';
import { ThemeProvider } from 'styled-components';

import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';

import { Repositiories } from '..';

import { THEMES } from '../../../theme/schema';

import { IItEachRepostioriesPage } from './types';

import {
  CONTAINER_TABLE_WRAPPER_DATA_TEST_ID,
  ERROR_MESSAGE_DATA_TEST_ID,
  LOADING_WRAPPER_DATA_TEST_ID
} from '../constants';

jest.mock('@apollo/client', () => ({
  useQuery: jest
    .fn()
    .mockReturnValueOnce({
      loading: true,
      error: undefined,
      rows: []
    })
    .mockReturnValueOnce({ loading: false, error: true, rows: [] })
    .mockReturnValueOnce({
      loading: false,
      error: undefined,
      rows: [
        {
          name: 'React',
          stars: 11111111111,
          forks: 22222222,
          id: '12313'
        }
      ]
    })
}));

afterAll(() => {
  jest.clearAllMocks();
  cleanup();
});

describe('Repositories page test', () => {
  describe('1. Checked the result of the repositories component with diffrent result of the query', () => {
    it.each<IItEachRepostioriesPage>([
      { numberTest: 1, dataTestId: LOADING_WRAPPER_DATA_TEST_ID },
      { numberTest: 2, dataTestId: ERROR_MESSAGE_DATA_TEST_ID },
      { numberTest: 3, dataTestId: CONTAINER_TABLE_WRAPPER_DATA_TEST_ID }
    ])('1.$numberTest. should render repositiories page with data-testid equal to: $dataTestId', ({ dataTestId }) => {
      const { container, unmount } = render(
        <ThemeProvider theme={THEMES.data.main}>
          <Repositiories />
        </ThemeProvider>
      );

      expect(container.innerHTML).toContain(dataTestId);

      unmount();
    });
  });
});
