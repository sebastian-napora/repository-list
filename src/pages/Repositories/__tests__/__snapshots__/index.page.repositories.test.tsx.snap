// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories page test 1. Checked the result of the repositories component with diffrent result of the query 1.1 should render repositiories page with data-testid equal to: loading-wrapper 1`] = `
.c0 {
  height: 80vh;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
  data-testid="loading-wrapper"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;

exports[`Repositories page test 1. Checked the result of the repositories component with diffrent result of the query 1.2 should render repositiories page with data-testid equal to: error-message 1`] = `
<div
  data-testid="error-message"
>
  Something went wrong, try again later.
</div>
`;

exports[`Repositories page test 1. Checked the result of the repositories component with diffrent result of the query 1.3 should render repositiories page with data-testid equal to: container-table-wrapper 1`] = `
.c1 {
  width: 100%;
}

.c0 {
  font-size: 22px;
  width: 100%;
  text-align: center;
}

<div
  class="MuiContainer-root MuiContainer-maxWidthSm"
  data-testid="container-table-wrapper"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <h1
      class="c0"
    >
      List of repositiories
    </h1>
    <div
      class="MuiFormControl-root MuiTextField-root c1"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
        for="outlined-basic"
        id="outlined-basic-label"
      >
        Search repository
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="outlined-basic"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Search repository
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-root-5 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiPaper-root makeStyles-root-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiTableContainer-root makeStyles-container-7"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root MuiTable-stickyHeader"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
              >
                Stars
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Forks
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                React
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                11111111111
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                22222222
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            id="mui-97015"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-97015 mui-13433"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              id="mui-13433"
              role="button"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-1 of 1
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
